module/x/gravity/keeper/batch_test.go

```diff
@@ -380,3 +380,49 @@ func TestEmptyBatch(t *testing.T) {
 
 	require.Nil(t, batchTx)
 }
+
+func TestBatchWithBlacklistedAddress(t *testing.T) {
+	input := CreateTestEnv(t)
+	ctx := input.Context
+	var (
+		now                 = time.Now().UTC()
+		mySender, _         = sdk.AccAddressFromBech32("cosmos1ahx7f8wyertuus9r20284ej0asrs085case3kn")
+		myTokenContractAddr = common.HexToAddress("0x429881672B9AE42b8EbA0E26cD9C73711b891Ca5") // Pickle
+		allVouchers         = sdk.NewCoins(
+			types.NewERC20Token(414, myTokenContractAddr).GravityCoin(),
+		)
+	)
+
+	// mint some voucher first
+	require.NoError(t, input.BankKeeper.MintCoins(ctx, types.ModuleName, allVouchers))
+	// set senders balance
+	input.AccountKeeper.NewAccountWithAddress(ctx, mySender)
+	require.NoError(t, fundAccount(ctx, input.BankKeeper, mySender, allVouchers))
+
+	amount := types.NewERC20Token(uint64(100), myTokenContractAddr).GravityCoin()
+	fee := types.NewERC20Token(0, myTokenContractAddr).GravityCoin()
+	// send to zero address blacklisted
+	_, err := input.GravityKeeper.createSendToEthereum(ctx, mySender, types.ZeroAddressString, amount, fee)
+	require.NoError(t, err)
```

Comment:
should `createSendToEthereum` error on blocked addresses?

Reply:
the blacklist is added to the msg server `SendToEthereum` but there is no check on `createSendToEthereum`